src/Compiler/Microsoft.AspNetCore.Razor.Language/test/TestFiles/IntegrationTests/ComponentDesignTimeCodeGenerationTest/RenderMode_Directive_FullyQualified/TestComponent.ir.txt

```diff
@@ -6,29 +6,28 @@
         UsingDirective - (97:4,1 [36] ) - global::System.Threading.Tasks
         UsingDirective - (136:5,1 [45] ) - global::Microsoft.AspNetCore.Components
         CSharpCode - 
-            IntermediateToken -  - CSharp - [global::Test.PrivateComponentRenderModeAttribute]
+            IntermediateToken -  - CSharp - [global::Test.TestComponent.PrivateComponentRenderModeAttribute]
         ClassDeclaration -  - public partial - TestComponent - global::Microsoft.AspNetCore.Components.ComponentBase - 
             DesignTimeDirective - 
-                DirectiveToken - (12:0,12 [53] x:\dir\subdir\Test\TestComponent.cshtml) - Microsoft.AspNetCore.Components.Web.RenderMode.Server
             CSharpCode - 
                 IntermediateToken -  - CSharp - #pragma warning disable 0414
             CSharpCode - 
                 IntermediateToken -  - CSharp - private static object __o = null;
             CSharpCode - 
                 IntermediateToken -  - CSharp - #pragma warning restore 0414
             MethodDeclaration -  - protected override - void - BuildRenderTree
-        ClassDeclaration -  - file sealed - PrivateComponentRenderModeAttribute - global::Microsoft.AspNetCore.Components.RenderModeAttribute - 
-            DesignTimeDirective - 
-            CSharpCode - 
-                IntermediateToken -  - CSharp - #pragma warning disable 0414
-            CSharpCode - 
-                IntermediateToken -  - CSharp - private static object __o = null;
-            CSharpCode - 
-                IntermediateToken -  - CSharp - #pragma warning restore 0414
-            CSharpCode - 
-                IntermediateToken -  - CSharp - private static global::Microsoft.AspNetCore.Components.IComponentRenderMode ModeImpl => 
-                CSharpCode - (12:0,12 [53] x:\dir\subdir\Test\TestComponent.cshtml)
-                    IntermediateToken -  - CSharp - Microsoft.AspNetCore.Components.Web.RenderMode.Server
-                IntermediateToken -  - CSharp - ;
-            CSharpCode - 
-                IntermediateToken -  - CSharp - public override global::Microsoft.AspNetCore.Components.IComponentRenderMode Mode => ModeImpl;
+            ClassDeclaration -  - private sealed - PrivateComponentRenderModeAttribute - global::Microsoft.AspNetCore.Components.RenderModeAttribute - 
+                DesignTimeDirective - 
```

Comment:
Also curious why there are empty `DesignTimeDirective`s, but perhaps that's unrelated to this PR.

Reply:
The code for adding the design time directives is unconditional. It always adds them, even if they have no children, we could fairly easily fix that though https://source.razor.fyi/#Microsoft.AspNetCore.Razor.Language/Extensions/DesignTimeDirectivePass.cs,75

Reply:
Filed https://github.com/dotnet/razor/issues/9259 to track